### PR TITLE
BLOCKS-35 - Parse Dropdown determined by Spinner [AK]

### PR DESCRIPTION
**Pull request by whati001**
Add spinner value support, fixed wrong value names in bricks. 

### Your checklist for this pull request
- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Make sure you use BLOCKS instead of CATROID in your commit message
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed (Actions)
- [x] Post a message in the *#catblocks* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: i18n/create_msg.js [hunk 1]
```diff
@@ -54,6 +54,15 @@ Blockly.CatblocksMsgs.setLocale = function(locale) {
...
 };
+
+Blockly.CatblocksMsgs.getCurrentLocale = function() {
+  return Blockly.CatblocksMsgs.currentLocale_;
+};
+
+Blockly.CatblocksMsgs.getCurrentLocaleValues = function() {
+  return Blockly.CatblocksMsgs.locales[Blockly.CatblocksMsgs.getCurrentLocale()];
+};
+
```
Comment (whati001) [inline i18n/create_msg.js]: new function needed to retrieve the current language values inside the parser

File: src/js/parser/parser.js [hunk 2]
```diff
@@ -260,6 +264,22 @@ function parseBrick(brick) {
...
+/**
+ * Return crowding value or default
+ * @param {*} key 
+ * @param {*} def 
+ */
+const getMsgValueOrDefault = (key, def = "---") => {
+  if (key === undefined) return def;
+  const msgValue = MESSAGES[key];
+  return msgValue ? msgValue : def;
+};
```
Comment (whati001) [inline src/js/parser/parser.js]: get value from language file

File: src/js/parser/parser.js [hunk 3]
```diff
@@ -268,34 +288,81 @@ const getNodeValueOrDefault = (node, def = "---") => {
...
+  case 'sceneToStart':
+  case 'objectToClone':
+  case 'soundName':
+  case 'motor':
+  case 'tone':
+  case 'eye':
+  case 'pointedObject':
+  case 'ledStatus':
+  case 'type':
+  case 'sceneForTransition': {
```
Comment (whati001) [inline src/js/parser/parser.js]: added missing spinner tags

File: src/js/render/file_dropper.js [hunk 4]
```diff
@@ -233,7 +233,7 @@ export class FileDropper {
           }
           
           const fd = FileDropper.getInstance();
-          fd.renderProgram(fd.share, fd.container, codeXML, containerfile.name, containerCounter, fileMap).then(result => {
+          fd.renderProgram(fd.share, fd.container, codeXML, containerfile.name, containerCounter, fileMap).then(() => {
```
Comment (whati001) [inline src/js/render/file_dropper.js]: remove eslint value

File: test/jsunit/block/block.test.js [hunk 5]
```diff
@@ -233,9 +233,10 @@ describe('WebView Block tests', () => {
         block.initSvg();
         block.render(false);
         //get default scale value
+        const refValue = Blockly.Bricks['ChangeBrightnessByNBrick'].args0[0].text;
         let value = block.inputList[0].fieldRow[1].getText();
         return (block.getFieldValue() === 'Change brightness  by'
-          && value === '50');
+          && value === refValue);
```
Comment (whati001) [inline test/jsunit/block/block.test.js]: better logic -> avoid hardcoding values in tests -< use definition in Blockly.Bricks

File: test/jsunit/parser/parser.test.js [hunk 6]
```diff
@@ -247,6 +247,47 @@ describe('Catroid to Catblocks parser tests', () => {
   });
 
 
+  describe('Spinner parsing test', () => {
+    /**
+      * Test if parser handels correct spinner value properly
```
Comment (whati001) [inline test/jsunit/parser/parser.test.js]: new tests

**Comment by Hiebl** [comment]
is there i programm i can test against?

**Comment by wslany** [comment]
PRAISE: Thanks!

**Pull request merged**